src/kili/services/import_csv/__init__.py

```diff
@@ -0,0 +1,141 @@
+"""Import from csv module."""
+import csv
+import logging
+from typing import Dict, List, Optional, Tuple
+
+from kili.services.types import LabelType
+
+logging.basicConfig()
+logger = logging.getLogger("kili.services.import_csv")
+logger.setLevel(logging.INFO)
+
+
+# pylint: disable=too-many-arguments,too-many-locals
+def import_csv(
+    kili,
+    csv_file: str,
+    content_column: str,
+    unique_categories: Optional[List[str]],
+    category_column: Optional[str],
+    external_id_column: Optional[str],
+    csv_separator: str,
+    label_type: LabelType,
+    project_id: Optional[str],
+    project_title: Optional[str],
+    project_description: Optional[str],
+) -> str:
+    """Import a csv file into a text classification project."""
+    logger.info(f"Reading csv file: {csv_file}")
+    content_array, categories_array, external_id_array = _read_csv(
+        csv_file, content_column, category_column, external_id_column, csv_separator
+    )
+
+    if project_id is None:
+        if project_title is None:
+            raise ValueError("`project_title` is required if `project_id` is not provided.")
+
+        if project_description is None:
+            raise ValueError("`project_description` is required if `project_id` is not provided.")
+
+        if unique_categories:
+            categories = unique_categories
+        elif categories_array:
+            categories = list(set(categories_array))
+        else:
+            raise ValueError(
+                "Cannot create the project if categories are not provided. You must provide either"
+                " `unique_categories` or `category_column`."
+            )
+
+        project_id = _create_project(kili, project_title, project_description, categories)
+
+        logger.info(f"Created classification project {project_title} with id {project_id}")
+
+    logger.info(f"Adding {len(content_array)} assets to project {project_id}")
+    kili.append_many_to_dataset(
```

Comment:
we do not call an entrypoint in a service (we cannot call a service either)

Reply:
if we can't call either entrypoint nor service code, aren't we just going to duplicate code in the end? not sure if it's worth it